### PR TITLE
Add User Registration with Ask Password and Email Verification tutorials

### PR DESCRIPTION
**Pull request by YvonneW**
Added the following `.md` files:
- docs/learn/user-registration.md 
- docs/learn/adding-email-special-characters.md
- docs/learn/user-regisrtation-with-email-verification.md
- docs/learn/user-registration-with-password-entry.md 
- docs/learn/managing-user-accounts.md

Added the following `.png` files that are associated with the above `.md` files: 
- docs/assets/img/learn/add-new-user-ask-password.png
- docs/assets/img/learn/add-new-user-define-password.png
- docs/assets/img/learn/add-new-user-option.png
- docs/assets/img/learn/add-new-user-screen.png
- docs/assets/img/learn/add-users-and-roles-menu-item.png
- docs/assets/img/learn/ask-password-email.png
- docs/assets/img/learn/ask-password-enable-email-verification.png
- docs/assets/img/learn/ask-password-expiry-time.png
- docs/assets/img/learn/claim-list-wso2.png
- docs/assets/img/learn/claims-list-menu-item.png
- docs/assets/img/learn/email-claim-edit.png
- docs/assets/img/learn/pickup-user-registration-with-email-verification.png
- docs/assets/img/learn/pickup-user-registration-with-password.png
- docs/assets/img/learn/reset-password-screen.png
- docs/assets/img/learn/resident-menu-item.png
- docs/assets/img/learn/user-onboarding-option.png
- docs/assets/img/learn/user-regisrtation-verification-email.png
- docs/assets/img/learn/wso2-claim-edit-screen.png

Updated the following `.md` files:
- mkdocs.yml
- docs/learn/creating-users-using-the-ask-password-option.md

### CODE REVIEW

File: en/docs/learn/adding-email-special-characters.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Configuring Emails with Special Characters
+
+Follow the steps below to accommodate email addresses with special character in WSO2 Identity Server.  
+
+!!! tip "Before you begin"
+
+	1.	[Run WSO2 Identity Sever](../../setup/running-the-product/).
+
+	2.	Sign in to the WSO2 Identity Server [Management Console](../../setup/getting-started-with-the-management-console/) at `https://<SERVER_HOST>:9443/carbon` as an administrator.
```
Comment (gomathyk) [inline en/docs/learn/adding-email-special-characters.md]: Sign in to WSO2 Identity Server

File: en/docs/learn/adding-email-special-characters.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+2.	Click  **http://wso2.org/claims**.
+
+	<img src="../../assets/img/learn/claim-list-wso2.png" alt="WSO2 Claim" width="600" style="border:1px solid grey">   	
+
+3.	Under **Email**, click **Edit**.
+
+	<img src="../../assets/img/learn/email-claim-edit.png" alt="Claim Edit option" width="600" style="border:1px solid grey"> 
+
+4.	Enter the characters you need out of the !#$%&'*+-=?^_ special characters to the **Regular Expression** text box.
```
Comment (gomathyk) [inline en/docs/learn/adding-email-special-characters.md]: into the **Regular Expression**

File: en/docs/learn/adding-email-special-characters.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+	<img src="../../assets/img/learn/wso2-claim-edit-screen.png" alt="WSO2 Claim Edit screen" width="700" style="border:1px solid grey"> 
+
+
+	!!! example "Using # and $"
+
+		-	To add **`#`** to the regex email pattern:
```
Comment (gomathyk) [inline en/docs/learn/adding-email-special-characters.md]: email regex pattern?

File: en/docs/learn/adding-email-special-characters.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+	!!! example "Using # and $"
+
+		-	To add **`#`** to the regex email pattern: 
+			```
+			^([a-zA-Z0-9_.-#])+\@(([a-zA-Z0-9#-])+.)+([a-zA-Z0-9#]{2,4})+$`
+			```			
+
+		-	To add **`$`** to the regex email pattern.
```
Comment (gomathyk) [inline en/docs/learn/adding-email-special-characters.md]: email regex pattern?

File: en/docs/learn/managing-user-accounts.md [hunk 5]
```diff
@@ -1,2 +1,16 @@
 # Managing User Accounts
 
+A **user account** is the digital representation of a physical user who interacts with an application. It is an entity that comprises the private information such as user name and password of a user.
```
Comment (gomathyk) [inline en/docs/learn/managing-user-accounts.md]: the user name and password

File: en/docs/learn/user-regisrtation-with-email-verification.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+	1.	To request email verification from the users who are registered with a default password, add the following configurations. 
+
+		```toml
+		[identity_mgt.user_onboarding]
+		enable_email_verification = true 
+		lock_on_creation=true
+		```
+
+	2.	To configure the email server to send emails requesting passwor entry, add the following configurations.
```
Comment (gomathyk) [inline en/docs/learn/user-regisrtation-with-email-verification.md]: a password entry

File: en/docs/learn/user-regisrtation-with-email-verification.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+5.	Under the **Account Management Policies** section, click **User Onboarding**.
+
+	<img src="../../assets/img/learn/user-onboarding-option.png" alt="User Onboarding Option" width="700" style="border:1px solid grey">  	
+
+6.	Select the **Enable User Email Verification** check box.
+
+	<img src="../../assets/img/learn/ask-password-enable-email-verification.png" alt="Ask password code expiry time text box" width="700" style="border:1px solid grey">  	  
+
+7.	You may enter the confirmation email validity period (in minutes) in the **Email verification code expiry time** text box.
```
Comment (gomathyk) [inline en/docs/learn/user-regisrtation-with-email-verification.md]: Shall we start this with "Enter the confirmation..." like the other instructions?

File: en/mkdocs.yml [hunk 8]
```diff
@@ -64,8 +64,13 @@ nav:
     - 'Tutorials':
       - 'Tutorials': learn/tutorials.md
       - 'Basic Tutorials':
-        - 'User Management': 
-          - 'Adding Users and Roles': learn/adding-users-and-roles.md
+        - 'Users and Roles': learn/adding-users-and-roles.md
+        - 'User Accounts': 
+          - 'User Accounts': learn/managing-user-accounts.md
+          - 'User Regisrtation':
```
Comment (gomathyk) [inline en/mkdocs.yml]: Registration spelling

File: en/mkdocs.yml [hunk 9]
```diff
@@ -64,8 +64,13 @@ nav:
...
+            - 'User Registration': learn/user-registration.md
+            - 'With Password Entry': learn/user-registration-with-password-entry.md
+            - 'With Email Verification': learn/user-regisrtation-with-email-verification.md
```
Comment (gomathyk) [inline en/mkdocs.yml]: user-registration-with-email-verification

File: en/docs/learn/user-regisrtation-with-email-verification.md (hunk 7 above)
Comment (YvonneW) [reply en/docs/learn/user-regisrtation-with-email-verification.md]: This is an optional step. Optional steps should start with, 'You may'.

File: en/mkdocs.yml (hunk 9 above)
Comment (YvonneW) [reply en/mkdocs.yml]: Fixed

**Pull request merged**